src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -132,7 +133,8 @@ protected override Tasks.Task DisposeCoreAsync(bool initialized)
         {
             foreach (var provider in _launchProviders.Value.Debuggers)
             {
-                if (await provider.Value.CanLaunchAsync(DebugLaunchOptions.DesignTimeExpressionEvaluation)
+                if (string.Equals(DebuggerName, provider.Metadata.DebuggerName, StringComparison.OrdinalIgnoreCase) &&
+                    await provider.Value.CanLaunchAsync(DebugLaunchOptions.DesignTimeExpressionEvaluation)
```

Comment:
The last time I debugged the code during the launch, I saw the value '0' and the `DesignTimeExpressionEvaluation` being passed but I couldn't figure who was passing those values. Hence I went with `DebugLaunchOptions.DesignTimeExpressionEvaluation`.

I checked all the current implementation of the interface and none of them use the `DebugLaunchOptions` flag for `CanLaunchAsync`. If we feel that this option is wrong, we can come back to change it.